packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -180,6 +180,17 @@ abstract class BoxBorder extends ShapeBorder {
       ..addRect(rect);
   }
 
+  @override
+  void paintInterior(Canvas canvas, Rect rect, Paint paint, { TextDirection? textDirection }) {
+    // If `ShapeDecoration(shape: Border.all())`, ShapeDecoration doesn't
```

Comment:
If -> For?

Maybe clarify this further, explaining that this class only ever paints a rounded border if `paint` is given one and that this is an extension to the superclass' API.

Reply:
Thanks! I re-wrote it:
```
    // For `ShapeDecoration(shape: Border.all())`, a rectangle with sharp edges
    // is always painted. There is no BorderRadius parameter for
    // ShapeDecoration or Border, only for BoxDecoration, which doesn't
    // call this method.
```